src/proposals.rs

```diff
@@ -123,9 +123,33 @@ async fn poll_approved_proposals(
 ) -> Result<(), Error> {
     let config = APP.config();
     let mut gov_client = GovQueryClient::connect(config.cosmos.grpc.clone()).await?;
+
+    // Tracks which pending proposals have been submitted already for skipping
+    let mut pending_already_processed = HashSet::<u64>::default();
+
+    // Used to track what ID to process when there are pending proposals
+    let mut last_checked_pending_id = 0;
+
+    // Whether there are pending proposals to check
+    let mut pending_proposals = false;
     loop {
         // Proposal IDs start at 1, so this should be ok even for the first query after startup.
-        let proposal_id = &state.last_processed_proposal_id + 1;
+        let proposal_id = if pending_proposals {
+            // remove any stale already-processed pending proposal ids from the set in case
+            // they stuck around from a previous pending-processing phase.
+            pending_already_processed.retain(|&x| x > state.last_processed_proposal_id);
```

Comment:
Given the overall amount of government proposals and how often Steward gets restarted, this should be a really negligible number, it's probably not even necessary to do.

Reply:
coo'